### DISCUSSION TITLE
Fheroes2 launch time

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#3337] [General]
I think I have a clue about the program launch time reported in this thread: https://heroes2.forumactif.com/t935-fheroes2-0-8-fheroes2-launch-time.

It seems that fheroes2.exe connects to the internet before launching and this may take some time.

If there is no internet connection available, the program launches without waiting.

If this is the case, it would be advisable to specify it, at least in the "README.txt" file, so that the user is informed.
And it should also be possible for the user to allow or not this connection, with an easily accessible setting in a file.

**Comment by ihhub** [comment]
Hi @LeHerosInconnu , this assumption is incorrect. The source code doesn't have anything related to networking.

**Reply by LeHerosInconnu** [reply]
Hello @ihhub,

Okay, thanks for the precison.
I'll keep looking for what might be causing this. :)

**Reply by ihhub** [reply]
Do you have any antivirus installed on your PC?

**Reply by LeHerosInconnu** [reply]
Hello @ihhub,

Only Microsoft Security Essentials.

**Reply by ihhub** [reply]
@LeHerosInconnu , could you please check in resource manager when fheroes2 startup takes so much time whether `Microsoft Security Essentials` eats a lot of CPU?

**Reply by LeHerosInconnu** [reply]
@ihhub.

I have done some tests.

It seems that the slowdown comes from the Real-time protection of Microsoft Security Essentials.
When Real-time protection is disabled the program (both windows) launches without delay.

**Reply by ihhub** [reply]
In this case we can close this topic 🙂

**Reply by LeHerosInconnu** [reply]
Okay, let's say it's closed.

**Comment by LeHerosInconnu** [comment]
I close this resolved discussion.
